src/main/java/demon/Commands.java

```diff
@@ -0,0 +1,285 @@
+     * Method takes in the tasks arraylist and a string.
+     *
+     * @param tasks an array containing all tasks.
+     * @param input String input by the user containing the description, date and time.
+     * @throws NoTimingException if date and time not given.
+     * @throws EmptyDescriptionException if description not given.
+     * @throws IOException if fails to write new task into file.
+     * @throws DateTimeParseException if date and time given is invalid.
+     */
+    public void addDeadline(TaskList tasks, String input)
+            throws NoTimingException,
+                   EmptyDescriptionException,
+                   IOException,
+                   DateTimeParseException {
+        String[] parts = input.split(" ",2);
+        if (parts.length < 2) {
+            throw new EmptyDescriptionException("Description is EMPTY!");
+        }
+        String deadline = parts[1];
+        String[] parts2 = deadline.split("/by");
+        if (parts2.length < 2) {
+            throw new NoTimingException("WOI! Please include deadline!");
+        }
+        String description = parts2[0].trim();
+        String by = parts2[1].trim();
+
+        // Automatically assume that if time is not given, then time is 0000hrs
+        int sizeOfBy = by.split(" ").length;
+        if (sizeOfBy < 2) {
+            by += " 0000";
+        }
+        // Format the date, time, and create Deadline object, add to list
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+        LocalDateTime dateTime = LocalDateTime.parse(by, formatter);
+        Deadline itemDeadline = new Deadline(description, dateTime);
+        tasks.getTaskList().add(itemDeadline);
+        String stringToSave = "D | " + (itemDeadline.getStatusIcon().equals("X") ? "1" : "0")
+                            + " | " + description + " | " + by + "\n";
+
+        // May produce IOException
+        storage.writeToFile(stringToSave);
+        Ui.printDivider();
+        Ui.showTaskAdded();
+        System.out.println(itemDeadline);
+        System.out.println("Now you have " + tasks.getTaskList().size() + " tasks in the list.");
+        Ui.printDivider();
+        Ui.promptNext();
+    }
+
+    /**
+     * Creates a To-do object as a task and stores into the taskList array.
+     * Method takes in the tasks arraylist and a string.
+     *
+     * @param tasks an array containing all tasks.
+     * @param input String input by the user containing the description.
+     * @throws EmptyDescriptionException if description not given.
+     * @throws IOException if fails to write new task into file.
+     */
+    public void addToDo(TaskList tasks, String input) throws EmptyDescriptionException, IOException {
+        String[] parts = input.split(" ",2);
+        if (parts.length < 2) {
+            throw new EmptyDescriptionException("Description is EMPTY!");
+        }
+        String toDo = parts[1];
+        Todo item_toDo = new Todo(toDo);
```

Comment:
Should 'item_toDo' be in `camelCase` withouth the underscore since it's a variable?